plugins/org.obeonetwork.m2doc/src/org/obeonetwork/m2doc/generator/TemplateProcessor.java

```diff
@@ -848,7 +861,7 @@ private void setGenericParameters(AbstractProviderClient templateProvider, Map<S
         EMap<String, Object> optionsMap = templateProvider.getOptionValueMap();
         Set<Entry<String, Object>> optionsMapEntries = optionsMap.entrySet();
         for (Entry<String, Object> optionsMapEntry : optionsMapEntries) {
-            if (optionTypes == null || optionTypes.get(optionsMapEntry.getKey()) == null) {
+            if ((optionTypes != null && optionTypes.get(optionsMapEntry.getKey()) == null) || optionTypes == null) {
```

Comment:
For your comment on "ExportRepresentationCommand" class, the test of refreshRepresentations is not needed because it is done in caller method but effectively for easy code reading it is better to add it. So I done it